examples/connext_dds/printing_qos/cs/QosPrintingExampleProgram.cs

```diff
@@ -0,0 +1,84 @@
+
+namespace QosPrintingExample
+{
+    /// <summary>
+    /// Runs the QosPrintingExample
+    /// </summary>
+    public static class HelloWorldProgram
+    {
+        /// <summary>
+        /// The Main function runs the Qos printing example.
+        /// </summary>
+        public static void Main(string[] args)
+        {
+            using DomainParticipant participant = DomainParticipantFactory.Instance.CreateParticipant(0);
+            Topic<HelloWorld> topic = participant.CreateTopic<HelloWorld>("Example QosPrintingExample_HelloWorld");
+            Publisher publisher = participant.CreatePublisher();
+
+            // In order to demonstrate a possible use-case for the Qos printing APIs
+            // we create a DataWriter which is a combination of policies set in
+            // the XML file, and of policies set in code.
+            DataWriterQos writerQos = QosProvider.Default.GetDataWriterQos()
+                .WithReliability(policy => policy.Kind = ReliabilityKind.BestEffort);
+            DataWriter<HelloWorld> writer = publisher.CreateDataWriter(topic, writerQos);
+
+
+            // The Qos we have obtained from the XML file inherits from the
+            // Generic::StrictReliable built-in profile. It also explicitly sets
+            // the Durability Qos policy. We have also explicitly set (in code)
+            // the Reliability Qos (overriding the ReliabilityKind set in the
+            // built-in profile). Using the Qos printing APIs we can check what
+            // is the final Qos in use by an entity.
+            Console.WriteLine("Printing DataWriterQos to stdout");
+            Console.WriteLine("Press enter to continue");
+            Console.ReadLine();
+            Console.WriteLine(writer.Qos);
+
+            // The output can be configured using QosPrintFormat
+            QosPrintFormat format = new QosPrintFormat
+            {
+                Indent = 1,
+                IsStandalone = true
+            };
+            Console.WriteLine("Printing formatted DataWriterQos to stdout");
+            Console.WriteLine("Press enter to continue");
+            Console.ReadLine();
+            Console.WriteLine(writer.Qos.ToString(format));
+
+            // By default, only the differences with respect to the documented
+            // QoS default (the default values as stated in the API reference)
+            // are printed. This behavior can be overridden by passing another
+            // QoS object to use as the baseQos to the ToString.
+            Console.WriteLine("Printing DataWriterQos with respect to \"BuiltinQosLib::Generic::StrictReliable\" to stdout");
+            Console.WriteLine("Press enter to continue");
+            Console.ReadLine();
+            Console.WriteLine(writer.Qos.ToString(
+                baseQos: QosProvider.Default.GetDataWriterQos("BuiltinQosLib::Generic.StrictReliable")));
+            
+            // Alternatively the sentinel value QosPrintAll.Value can be provided,
```

Comment:
To print all the QoS policies (not just the differences), call ToString with the sentinel QosPrintAll.Value